ydb_dbapi/connections.py

```diff
@@ -0,0 +1,351 @@
+            ),
+            IsolationLevel.SERIALIZABLE: IsolationSettings(
+                ydb.QuerySerializableReadWrite(), interactive=True
+            ),
+            IsolationLevel.ONLINE_READONLY: IsolationSettings(
+                ydb.QueryOnlineReadOnly(), interactive=True
+            ),
+            IsolationLevel.ONLINE_READONLY_INCONSISTENT: IsolationSettings(
+                ydb.QueryOnlineReadOnly().with_allow_inconsistent_reads(),
+                interactive=True,
+            ),
+            IsolationLevel.STALE_READONLY: IsolationSettings(
+                ydb.QueryStaleReadOnly(), interactive=True
+            ),
+            IsolationLevel.SNAPSHOT_READONLY: IsolationSettings(
+                ydb.QuerySnapshotReadOnly(), interactive=True
+            ),
+        }
+        ydb_isolation_settings = ydb_isolation_settings_map[isolation_level]
+        if self._tx_context and self._tx_context.tx_id:
+            raise InternalError(
+                "Failed to set transaction mode: transaction is already began"
+            )
+        self._tx_mode = ydb_isolation_settings.ydb_mode
+        self.interactive_transaction = ydb_isolation_settings.interactive
+
+    def get_isolation_level(self) -> str:
+        if self._tx_mode.name == ydb.QuerySerializableReadWrite().name:
+            if self.interactive_transaction:
+                return IsolationLevel.SERIALIZABLE
+            return IsolationLevel.AUTOCOMMIT
+        if self._tx_mode.name == ydb.QueryOnlineReadOnly().name:
+            if self._tx_mode.settings.allow_inconsistent_reads:
+                return IsolationLevel.ONLINE_READONLY_INCONSISTENT
+            return IsolationLevel.ONLINE_READONLY
+        if self._tx_mode.name == ydb.QueryStaleReadOnly().name:
+            return IsolationLevel.STALE_READONLY
+        if self._tx_mode.name == ydb.QuerySnapshotReadOnly().name:
+            return IsolationLevel.SNAPSHOT_READONLY
+        msg = f"{self._tx_mode.name} is not supported"
+        raise NotSupportedError(msg)
+
+    def cursor(self) -> Cursor | AsyncCursor:
+        if self._session is None:
+            raise RuntimeError("Connection is not ready, use wait_ready.")
+
+        if self.interactive_transaction:
+            self._tx_context = self._session.transaction(self._tx_mode)
+        else:
+            self._tx_context = None
+
+        self._current_cursor = self._cursor_cls(
+            session=self._session,
+            tx_context=self._tx_context,
+            autocommit=(not self.interactive_transaction),
+        )
+        return self._current_cursor
+
+
+class Connection(BaseConnection):
+    _driver_cls = ydb.Driver
+    _pool_cls = ydb.QuerySessionPool
+    _cursor_cls = Cursor
+
+    _driver: ydb.Driver
+    _pool: ydb.QuerySessionPool
+    _current_cursor: Cursor | None = None
+
+    def wait_ready(self, timeout: int = 10) -> None:
+        try:
+            self._driver.wait(timeout, fail_fast=True)
```

Comment:
**issue**: If shared_pool is used you should not wait for the driver availability, just check that driver is started